flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -370,45 +370,24 @@ public void onContainersCompleted(final List<ContainerStatus> statuses) {
 	@Override
 	public void onContainersAllocated(List<Container> containers) {
 		runAsync(() -> {
```

Comment:
Is there possibly any value in moving part of this method out of the async block? (removing requests and returning containers)

The idea being that under heavy loadcthe processing of the allocation could be delayed for quite a while.

Reply:
For this, we need to read `numPendingContainerRequests` which can only be done from within the actor's main thread. Hence without introducing some `volatiles` and potential race conditions I don't see an easy way.